libbeat/cmd/export/dashboard.go

```diff
@@ -58,16 +61,39 @@ func GenDashboardCmd(name, idxPrefix, beatVersion string) *cobra.Command {
 				os.Exit(1)
 			}
 
-			result, err := client.GetDashboard(dashboard)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error getting dashboard: %+v\n", err)
-				os.Exit(1)
+			// Export dashboards from yml file
+			if yml != "" {
+				results, info, err := dashboards.ExportAllFromYml(client, yml)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error getting dashboards from yml: %+v\n", err)
+					os.Exit(2)
+				}
+				for i, r := range results {
+					err = dashboards.SaveToFile(r, info.Dashboards[i].File, filepath.Dir(yml), client.GetVersion())
+					if err != nil {
+						fmt.Fprintf(os.Stderr, "Error saving dashboard '%s' to file '%s' : %+v\n",
+							info.Dashboards[i].ID, info.Dashboards[i].File, err)
+						os.Exit(2)
```

Comment:
Out of curiosity why you went with Exit(2)? I thought it was generally used for not correctly used CLI arguments.

Reply:
Nice catch. It was a tyop which I have copy-pasted. :man_facepalming: